--- a/game/server/ai_basenpc.cpp
+++ b/game/server/ai_basenpc.cpp
@@ -27,6 +27,9 @@
 #include "ai_interactions.h"
 #include "hl2_gamerules.h"
 #endif // HL2_DLL
+#ifdef EZ
+#include "sprite.h"
+#endif
 
 #include "ai_network.h"
 #include "ai_networkmanager.h"
@@ -583,6 +586,9 @@
 	}
 
 	Wake( false );
+#ifdef EZ
+	KillSprites( 0.0f );
+#endif
 	
 	//Adrian: Select a death pose to extrapolate the ragdoll's velocity.
 	SelectDeathPose( info );
@@ -1184,9 +1190,12 @@
 
 	if ( subInfo.GetDamage() >= 1.0 && !(subInfo.GetDamageType() & DMG_SHOCK ) )
 	{
+#ifndef EZ
 		if( !IsPlayer() || ( IsPlayer() && g_pGameRules->IsMultiplayer() ) )
+#endif
 		{
 			// NPC's always bleed. Players only bleed in multiplayer.
+			// EZ: Bad Cop also bleeds in singleplayer.
 			SpawnBlood( ptr->endpos, vecDir, BloodColor(), subInfo.GetDamage() );// a little surface blood.
 		}
 
@@ -10691,6 +10700,9 @@
 	DEFINE_KEYFIELD( m_iszEnemyFilterName,		FIELD_STRING, "enemyfilter" ),
 	DEFINE_FIELD( m_bImportanRagdoll,			FIELD_BOOLEAN ),
 	DEFINE_FIELD( m_bPlayerAvoidState,			FIELD_BOOLEAN ),
+#ifdef EZ
+	DEFINE_KEYFIELD( m_bNoGlow,					FIELD_BOOLEAN, "noglow" ),
+#endif
 
 	// Satisfy classcheck
 	// DEFINE_FIELD( m_ScheduleHistory, CUtlVector < AIScheduleChoice_t > ),
@@ -10866,6 +10878,12 @@
 	m_ScheduleHistory.RemoveAll();
 #endif//AI_MONITOR_FOR_OSCILLATION
 
+#ifdef EZ
+	if ( IsAlive() )
+	{
+		StartEye();
+	}
+#endif
 }
 
 void CAI_BaseNPC::Precache( void )
@@ -11303,6 +11321,10 @@
 	m_pSenses = NULL;
 	m_pPathfinder = NULL;
 	m_pLocalNavigator = NULL;
+#ifdef EZ
+	m_pEyeGlow = NULL;
+	m_bNoGlow = false;
+#endif
 
 	m_pSchedule = NULL;
 	m_IdealSchedule = SCHED_NONE;
@@ -11402,8 +11424,68 @@
 //-----------------------------------------------------------------------------
 // Purpose:
 //-----------------------------------------------------------------------------
+#ifdef EZ
+void CAI_BaseNPC::StartEye( void )
+{
+	if ( m_bNoGlow )
+		return;
+
+	for ( int i = 0; i < GetNumGlows(); i++ )
+	{
+		EyeGlow_t *glowData = GetEyeGlowData( i );
+		if ( glowData == NULL )
+			continue;
+
+		CSprite *sprite = GetGlowSpritePtr( i );
+		if ( sprite == NULL )
+		{
+			sprite = CSprite::SpriteCreate( glowData->spriteName, GetLocalOrigin(), false );
+			sprite->SetAttachment( this, LookupAttachment( glowData->attachment ) );
+			sprite->SetTransparency( glowData->renderMode, glowData->red, glowData->green, glowData->blue, glowData->alpha, kRenderFxNoDissipation );
+			sprite->SetColor( glowData->red, glowData->green, glowData->blue );
+			if ( glowData->brightness > 0 )
+				sprite->SetBrightness( glowData->brightness, 0.1f );
+			if ( glowData->scale > 0 )
+				sprite->SetScale( glowData->scale, 0.1f );
+			if ( glowData->proxyScale > 0 )
+				sprite->SetGlowProxySize( glowData->proxyScale );
+			sprite->SetAsTemporary();
+			SetGlowSpritePtr( i, sprite );
+		}
+	}
+}
+
+void CAI_BaseNPC::KillSprites( float flDelay )
+{
+	for ( int i = 0; i < GetNumGlows(); i++ )
+	{
+		CSprite *sprite = GetGlowSpritePtr( i );
+		if ( sprite )
+			sprite->FadeAndDie( flDelay );
+		SetGlowSpritePtr( i, NULL );
+	}
+}
+
+CSprite *CAI_BaseNPC::GetGlowSpritePtr( int i )
+{
+	if ( i != 0 )
+		return NULL;
+	return m_pEyeGlow;
+}
+
+void CAI_BaseNPC::SetGlowSpritePtr( int i, CSprite *sprite )
+{
+	if ( i != 0 )
+		return;
+	m_pEyeGlow = sprite;
+}
+#endif
+
 void CAI_BaseNPC::UpdateOnRemove(void)
 {
+#ifdef EZ
+	KillSprites( 0.0f );
+#endif
 	if ( !m_bDidDeathCleanup )
 	{
 		if ( m_NPCState == NPC_STATE_DEAD )
